src/wrench/resources.py

```diff
@@ -63,6 +63,13 @@ def decrypt_resource(resource: Resource, gpg: GPG) -> Resource:
     return resource._replace(secret=utils.decrypt(resource.encrypted_secret, gpg))
 
 
+def decrypt_resource_secret(resource_secret: ResourceSecret, gpg: GPG) -> ResourceSecret:
```

Comment:
I would just return the decrypted secret instead of modifying the `ResourceSecret` object and returning it. That means I would also remove the `secret` field from the `ResourceSecret` entity.